python_ta/checkers/structure_test_checker.py

```diff
@@ -0,0 +1,56 @@
+"""
+checker for using constant value in a conditional statement
+"""
+import astroid
+
+from pylint.interfaces import IAstroidChecker
+from pylint.checkers import BaseChecker
+from pylint.checkers.utils import check_messages
+
+
+class UsingStructureTestChecker(BaseChecker):
+
+    __implements__ = IAstroidChecker
+
+    name = 'using-constants-test'
+
+    msgs = {'W7778': ('using a structure which returns a constant value in a conditional statement'
+                      , 'using-constants-test'
+                      , 'Conditional statements should depend on a variable not a constant value.'
+                        'This is usually not what the user intended to do'),}
+
+    # this is important so that your checker is executed before others
+    priority = -1
+
+    def _check_all_constants(self, node):
```

Comment:
This code duplication isn't good. In fact, for this checker *only* the first check (`List`, `Tuple`, etc.) is necessary; the other checks should *not* occur.